### PR TITLE
Clean up root directory

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

**Comment by hayd** [comment]
Another thing is roll .eslintignore .eslintrc .prettierrc etc into package.json

**Comment by ry** [comment]
@hayd That would be amazing - I hate all these random configuration files.

File: .gitignore [hunk 1]
```diff
@@ -4,9 +4,13 @@
...
 /gh-pages/
 /target/
-/website/*.bundle.js
-/website/data.json
-/website/recent.json
+
+# We use something stronger than lockfiles, we have all NPM modules stored in a
+# git. We do not download from NPM during build.
+# https://github.com/denoland/deno_third_party
+yarn.lock
```
Comment (piscisaureus) [inline .gitignore]: I'd prefer to keep this actually

File: cli/tests/symlink_to_subdir [hunk 2]
```diff
@@ -0,0 +1 @@
+subdir/
```
Comment (piscisaureus) [inline cli/tests/symlink_to_subdir]: Add this path to .gitattributes and add a "symlink=dir" attribute.

File: .gitignore (hunk 1 above)
Comment (ry) [reply .gitignore]: Why? Also if you want to keep it, we can keep it in third_party - doesn't need to be in root.

Comment (piscisaureus) [reply .gitignore]: Actually I don't care enough to make the case for it.

File: cli/tests/symlink_to_subdir (hunk 2 above)
Comment (ry) [reply cli/tests/symlink_to_subdir]: Done

**Pull request merged**